### PR TITLE
Move pointwise log likelihood data to log_likelihood group (Pyro + NumPyro)

### PR DESCRIPTION
**Pull request by VincentBt**
Follows #796, for Pyro and NumPyro: I added "log_likelihood" to InferenceData, and added tests for multiple observed variables. 

Thanks for a Pyro/NumPyro expert to review this (@fehiepsi maybe?). Indeed, I am not entirely sure about the Pyro part: the test_multiple_observed_rv fails if y2 and y1 have different dimensions. 
For the rest, I copied io_pymc3.py, in particular for coord_name and dims.

Sorry in advance for the possible mistakes, it's my first commit.

### CODE REVIEW

File: arviz/tests/test_data_numpyro.py [hunk 1]
```diff
@@ -25,10 +26,10 @@ class Data:
     def get_inference_data(self, data, eight_schools_params):
         posterior_samples = data.obj.get_samples()
         model = data.obj.sampler.model
-        posterior_predictive = Predictive(model, posterior_samples).get_samples(
+        posterior_predictive = Predictive(model, posterior_samples).__call__(
```
Comment (fehiepsi) [inline arviz/tests/test_data_numpyro.py]: nit: `.__call__` is not required

File: arviz/tests/test_data_numpyro.py [hunk 2]
```diff
@@ -25,10 +26,10 @@ class Data:
...
             PRNGKey(1), eight_schools_params["J"], eight_schools_params["sigma"]
         )
-        prior = Predictive(model, num_samples=500).get_samples(
+        prior = Predictive(model, num_samples=500).__call__(
```
Comment (fehiepsi) [inline arviz/tests/test_data_numpyro.py]: ditto: `.__call__` is not required

File: arviz/tests/test_data_numpyro.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import pytest
 from jax.random import PRNGKey
 from numpyro.infer import Predictive
+import jax.numpy
```
Comment (fehiepsi) [inline arviz/tests/test_data_numpyro.py]: nit: this import is not required I guess?

**Comment by OriolAbril** [comment]
Can you update also Azure configuration file (in the same way as Travis) and add this to the changelog?

Everything else looks good, after these last changes we can merge

**Comment by VincentBt** [comment]
> In case the try except must be a block as it is now, the `pass` should probably be a `return None`

You're right, I changed the code as you suggested.

Ready to merge.

PS: is it possible to run the Travis and Azure checks on my computer for the specific files that changed? That would have decreased the number of unnecessary commits (not talking about having to wait for all the checks to finish on this page before getting a failure)

**Comment by OriolAbril** [comment]
Travis and Azure mainly run the tests with pytest and the linters. If you have the required dependencies installed, you can run `pytest arviz/tests/test_file.py` or `pytest arviz/tests -k "key1 and key2 or key3"` to tests only a subset of the files/test functions. In addition, running `.scripts/lint.sh` should mimic the lint job. 

If you want to keep special versions in your environment you can always use docker to run tests locally being sure the dependencies installed will be the same as the ones in Azure and Travis first job. There is a section on this in the [contributing guide](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#developing-in-docker). The `--test` flag is a shortcut to run all tests, to run only some of them, enter a bash session in docker and run the specific pytest command. 

Finally, the documentation is also generated automatically and sphinx (or sphinx inside docker) can be used to build it locally and check everything works. In many cases it may not be necessary.

**Comment by OriolAbril** [comment]
Thanks for the PR!

**Pull request merged**